src/Controller.sol

```diff
@@ -0,0 +1,205 @@
+	{
+		return setController(_controller);
+	}
+
+	function setController(Controller _controller) public onlyControllerCaller returns(bool)
+	{
+		controller = _controller;
+		return true;
+	}
+
+	function suicideFunds(address _target) public onlyControllerCaller returns(bool)
+	{
+		selfdestruct(_target);
+		return true;
+	}
+}
+
+contract Controller
+{
+	address public owner;
+	mapping(address => bool) public whitelist;
+	mapping(bytes32 => address) public registry;
+	bool public stopped = false;
+
+	modifier onlyWhitelistedCallers
+	{
+		assertIsWhitelisted(msg.sender);
+		_;
+	}
+
+	modifier devModeOwnerOnly
+	{
+		require(msg.sender == owner);
+		require(whitelist[owner]);
+		_;
+	}
+
+	modifier onlyOwnerCaller
+	{
+		require(msg.sender == owner);
+		_;
+	}
+
+	modifier onlyInBadTimes
+	{
+		require(stopped);
+		_;
+	}
+
+	modifier onlyInGoodTimes
+	{
+		require(!stopped);
+		_;
+	}
+
+	function Controller()
+	{
+		owner = msg.sender;
+		whitelist[owner] = true;
+	}
+
+	/*
+	 * Whitelisting [whitelisted augur contracts and dev mode can use it]
+	 */
+
+	function addToWhitelist(address _target) public onlyWhitelistedCallers returns(bool)
+	{
+		whitelist[_target] = true;
+		return true;
+	}
+
+	function removeFromWhitelist(address _target) public onlyWhitelistedCallers returns(bool)
+	{
+		whitelist[_target] = false;
+		return true;
+	}
+
+	function assertIsWhitelisted(address _target) public returns(bool)
+	{
+		require(whitelist[_target]);
+		return true;
+	}
+
+	/*
+	 * Registry for lookups [whitelisted augur contracts and dev mode can use it]
+	 */
+
+	function setValue(bytes32 _key, address _value) public onlyWhitelistedCallers returns(bool)
+	{
+		registry[_key] = _value;
+		return true;
+	}
+
+	function lookup(bytes32 _key) public returns(address)
+	{
+		return registry[_key];
+	}
+
+	function assertOnlySpecifiedCaller(address _caller, bytes32 _allowedCaller) public returns(bool)
+	{
+		require(registry[_allowedCaller] == _caller || (msg.sender == owner && whitelist[owner]));
+		return true;
+	}
+
+	/*
+	 * Contract Administration [dev mode can use it]
+	 */
+
+	function suicide(Controlled _target, address _destination) public devModeOwnerOnly returns(bool)
+	{
+		_target.suicideFunds(_destination);
+		return true;
+	}
+
+	function updateController(Controlled _target, Controller _newController) public devModeOwnerOnly returns(bool)
+	{
+		_target.setController(_newController);
+		return true;
+	}
+
+	 /*
+	 * Controller Administration [dev can transfer ownership anytime, mode can only switched from dev mode -> decentralized]
+	 */
+
+	function transferOwnership(address _newOwner) onlyOwnerCaller returns(bool)
+	{
+		// if in dev mode, blacklist old owner and whitelist new owner
+		if (whitelist[owner])
+		{
+			whitelist[owner] = false;
+			whitelist[_newOwner] = true;
+		}
+		owner = _newOwner;
+		return true;
+	}
+
+	function switchModeSoOnlyEmergencyStopsAndEscapeHatchesCanBeUsed() public devModeOwnerOnly returns(bool)
```

Comment:
This function will only ever be called once in the life of Augur, so its super-verbose-name doesn't bother me much (so I haven't put much thought into it).  However, if someone can offer a suggestion for something more pithy I am definitely open to it.  :)